lib/galaxy/webapps/base/controller.py

```diff
@@ -1249,6 +1249,7 @@ def _import_shared_workflow(self, trans, stored):
         workflow.stored_workflow = imported_stored
         imported_stored.latest_workflow = workflow
         imported_stored.user = trans.user
+        imported_stored.copy_tags_from(stored.user, stored)
```

Comment:
How do you feel about ripping the import route out of the client and use the API endpoint instead (https://github.com/galaxyproject/galaxy/blob/067d0b2592e742704ec3d14f200e26413ce369cb/lib/galaxy/webapps/galaxy/api/workflows.py#L419) ? Having this parallel code is quite the maintenance burden.

Reply:
👍 I can give it a try!

Reply:
Yeah .. but I see now that this is shared between the API controller and the web controller, so the situation is actually not that bad. We should still do it eventually, but it's not a priority.